Truncate(dev-guide): Added steps for truncate (#51)
fivetran/fivetran_sdk: development-guide.md

#### DescribeTable
This operation should report all columns in the destination table, including Fivetran system columns such as `_fivetran_synced` and `_fivetran_deleted`. It should also provide other additional information as applicable such as data type, `primary_key` and `DecimalParams`.

#### TruncateRequest
- This operation might be requested for a table that does not exist in the destination. In that case, it should NOT fail, simply ignore the request and return `success = true`.

> - It contains an optional `SoftTruncate` field which either contains a string column `deleted_column` or boolean `history_mode`. If `deleted_column` is present, then current sync mode is `soft-delete-mode` and if `history_mode` is true then current sync mode is `history-mode`, otherwise the current sync mode is `live-mode`.

Reviewer: ```suggestion
- `SoftTruncate`: This optional field can be configured to either regular `soft-truncate` mode by setting the `deleted_column` field or history `soft-truncate` mode by setting the `history_mode` field to TRUE. Both modes keep the rows in the destination and update relevant system columns to mark any matching rows as deleted. If optional `SoftTruncate` field is not configured, then perform a hard-truncate operation where matching rows are actually removed from the destination.
```

> - `SoftTruncate`: This optional field can be configured to either regular `soft-truncate` mode by setting the `deleted_column` field or history `soft-truncate` mode by setting the `history_mode` field to TRUE. Both modes keep the rows in the destination and update relevant system columns to mark any matching rows as deleted. If optional `SoftTruncate` field is not configured, then perform a `hard-truncate` operation where matching rows are actually removed from the destination.

Reviewer 2: ```suggestion
- `SoftTruncate`: This optional field can be configured to either of the modes:
  - To the regular `soft-truncate` mode - by setting the `deleted_column` field to TRUE
  - To the history `soft-truncate` mode - by setting the `history_mode` field to TRUE. 
  Both modes keep the rows in the destination and update the relevant system columns to mark any matching rows as deleted. If the optional `SoftTruncate` field is not configured, then perform a `hard-truncate` operation where matching rows are actually removed from the destination.
```

> - It contains `utc_delete_before` timestamp field which has millisecond precision and use it to DELETE/UPDATE the rows in destination table as per the sync mode:

Reviewer: ```suggestion
- `utc_delete_before`: This timestamp field has millisecond precision and used to DELETE/UPDATE the rows in destination table as per the following:
```

Author: incorporated

>   - **Soft-Delete Mode:** Set `deleted_column` to TRUE where `_fivetran_synced` column value is less than `utc_delete_before`.

Reviewer: ```suggestion
  - **Soft-Truncate:** Set `deleted_column` to TRUE where `_fivetran_synced` column value is less than `utc_delete_before`.
```

Author: incorporated

> - `utc_delete_before`: This timestamp field has millisecond precision and used to DELETE/UPDATE the rows in destination table as per the following:

Reviewer 2: ```suggestion
- `utc_delete_before`: This timestamp field has millisecond precision. It is used to DELETE/UPDATE the rows in destination table depending on the configured mode:
```

>   - **Soft-Truncate:** Set `deleted_column` to TRUE where `_fivetran_synced` column value is less than `utc_delete_before`.

Reviewer 2: What is the actor of the set and delete actions? Is it on the user to implement or it's done automatically? In the latter case, I believe the passive voice should be a better option, like "column is set" etc.

>   - **History Mode:** Set `_fivetran_active` column value to FALSE, `_fivetran_end` column value to `utc_delete_before` value, where `_fivetran_synced` column value is less than `utc_delete_before` and `_fivetran_active` is TRUE.

Reviewer: ```suggestion
  - **History Truncate:** Set `_fivetran_active` column value to FALSE, `_fivetran_end` column value to `utc_delete_before` value, where `_fivetran_synced` column value is less than `utc_delete_before` and `_fivetran_active` is TRUE.
```

Author: incorporated

>   - **Live Mode:** DELETE the record where `_fivetran_synced` column value is less than `utc_delete_before`

Reviewer: ```suggestion
  - **Hard Truncate:** DELETE the record where `_fivetran_synced` column value is less than `utc_delete_before`
```

Author: incorporated
